HH-437 | Remove hobby type, reorganize search UI elements, fix age selector's font size & color

karisal-anders opened the pull request:
## Description

Remove hobby type everywhere.
Reorganize hobbies-helsinki's search UI elements and fix its age selector's font size & color.

See commit messages for details.

The **search** UI's **element** **organization** was **approved** by the **product owner**—
we talked about the widest mode only.

## Related

[HH-437](https://helsinkisolutionoffice.atlassian.net/browse/HH-437)

## Testing

### Automated tests

### Manual testing

## Screenshots

### Widest

<img width="1537" height="399" alt="image" src="https://github.com/user-attachments/assets/dd72f2f8-589a-429d-92a3-f34b8448ae20" />

## Middle width

<img width="1198" height="394" alt="image" src="https://github.com/user-attachments/assets/ddec5a05-5f52-40c5-9553-69b031b39358" />

## Second to narrow

<img width="963" height="467" alt="image" src="https://github.com/user-attachments/assets/95a6282c-d29f-4275-9924-3db7417e1f5c" />

## Narrowest

<img width="591" height="700" alt="image" src="https://github.com/user-attachments/assets/c3f38013-879b-4162-9ade-7c067a067988" />

## Additional notes


[HH-437]: https://helsinkisolutionoffice.atlassian.net/browse/HH-437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

karisal-anders committed:
feat: remove hobby type completely from codebase & clarify similar use

done:
- remove hobby type completely from codebase everywhere
- clarify similar uses in events-helsinki and sports-helsinki apps,
  i.e. sports uses sports categories, events uses event categories
- fix linting issues

refs HH-437 (hobby type is unneeded)

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr829.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders committed:
feat(hobbies): reorganize search menu elements after hobby type removal

the full width view's organization was approved by the product owner

refs HH-437

karisal-anders committed:
fix(components): make rangeDropdown's font size & color like others

This makes the font size and font color in hobbies-helsinki's search age
selector be consistent with the other search elements.

NOTE:
As RangeDropdown is not used anywhere else than in hobbies-helsinki app
this does not affect sports or events apps currently.

refs HH-437

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr829.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders commented:
Rebased on current main and resolved merge conflict.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr829.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela commented on apps/hobbies-helsinki/README.md:
```diff
@@ -307,7 +307,7 @@ The "main events search" component is used in the search page at `/search`.
 | keywordAnd        | []                                       | x            | -        | Used for "Only children events" when it will be set to "yso:p4354"                            |
 | keywordOrSet2     | []                                       | x            | -        | A list of keywords (e.g categories or premapped categories)                                   |
-| keywordOrSet3     | []                                       | x            | -        | A list of keywords (e.g hobby types or premapped hobby types)                                 |
+| keywordOrSet3     | []                                       | x            | -        | A list of keywords                                                                            |
```
Is the `keywordOrSet3` used at all? I think it should be removed from this table.

nikomakela replied:
https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/829/files#r2367141906

karisal-anders replied:
It's better to remove it completely.

karisal-anders replied:
Removed in [force push](https://github.com/City-of-Helsinki/events-helsinki-monorepo/compare/0c4b93bf7a97cfcf9931e26e574277fea7a8bcac..c3d7751230097612c20b6a981ca08511b1826dd9).

nikomakela commented on apps/hobbies-helsinki/src/domain/search/eventSearch/utils.tsx:
```diff
@@ -268,7 +230,7 @@ export const getEventSearchVariables = ({
     // internetBased: onlyRemoteEvents || undefined,
     keywordOrSet2: [...(keyword ?? []), ...mappedCategories],
-    keywordOrSet3: [...(keyword ?? []), ...mappedHobbyTypes],
+    keywordOrSet3: [...(keyword ?? [])],
```
```
   keywordOrSet2: [...(keyword ?? []), ...mappedCategories],
   keywordOrSet3: [...(keyword ?? [])],
```

If the keyword is already included in the keywordOrSet2, it should not make any difference to list it in a another keywordOrSet. It would just probably make the query a bit slower but keep the result set the same.

karisal-anders replied:
It's better to remove it completely.

karisal-anders replied:
Removed in [force push](https://github.com/City-of-Helsinki/events-helsinki-monorepo/compare/0c4b93bf7a97cfcf9931e26e574277fea7a8bcac..c3d7751230097612c20b6a981ca08511b1826dd9).

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr829.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr829.dev.hel.ninja :laughing::tada::tada::tada:
